### PR TITLE
added all files to implement mlp model

### PR DESCRIPTION
**Pull request by adil-a**
Added the necessary files for benchmarking an MLP model

### CODE REVIEW

File: torchbenchmark/models/mlp/__init__.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+class Model(BenchmarkModel):
+    task = OTHER.OTHER_TASKS
+
+    # same hyperparameters as found on sklearn:
+    # https://scikit-learn.org/stable/modules/generated/sklearn.neural_network.MLPClassifier.html
+    def __init__(self, device='cpu', jit=False, weight_decay=1e-4, lr=1e-3, beta_1=0.9, beta_2=0.999, eps=1e-8,
+                 *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.device = device
+        self.jit = jit
```
Comment (GeraldCSC) [inline torchbenchmark/models/mlp/__init__.py:27]: it seems like when jit=False we should be raising NotImplementedError as seen in this file https://github.com/pytorch/benchmark/blob/af6305c43fc6016c4893412d9604359a80719cdc/torchbenchmark/models/maml/__init__.py#L80

File: torchbenchmark/models/mlp/__init__.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        if self.device == 'cuda':
+            torch.backends.cudnn.deterministic = False
+            torch.backends.cudnn.benchmark = False
```
Comment (GeraldCSC) [inline torchbenchmark/models/mlp/__init__.py:30]: my bad, on cluster inspection this seems to only affect convolution algorithms which is not relevant here, so safe to remove

File: torchbenchmark/models/mlp/__init__.py [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+        X, y = datasets.load_breast_cancer(return_X_y=True)
+        X_train, X_val, y_train, y_val = train_test_split(X, y, random_state=1337)
```
Comment (GeraldCSC) [inline torchbenchmark/models/mlp/__init__.py:33]: how about `map(lambda x: torch.from_numpy(x.astype(np.float32)), train_test_split(X, y, random_state=1337))`

File: torchbenchmark/models/mlp/__init__.py [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+        X_train, X_val, y_train, y_val = X_train.astype(np.float32), X_val.astype(np.float32), \
+                                         y_train.astype(np.float32), y_val.astype(np.float32)
+        X_train, X_val, y_train, y_val = torch.from_numpy(X_train), torch.from_numpy(X_val), \
+                                         torch.from_numpy(y_train), torch.from_numpy(y_val)
```
Comment (GeraldCSC) [inline torchbenchmark/models/mlp/__init__.py:37]: how about `map(lambda x: torch.from_numpy(x.astype(np.float32)), train_test_split(X, y, random_state=1337))`

File: torchbenchmark/models/mlp/__init__.py [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+        for _ in range(niter):
+            out = self.model(self.X_eval)
+
+    def get_module(self):
+        return self.model, self.X_train
+
+
+if __name__ == '__main__':
+    for device in ['cpu', 'cuda']:
+        print("Testing device {}, JIT {}".format(device, 'False'))
```
Comment (GeraldCSC) [inline torchbenchmark/models/mlp/__init__.py:76]: no need to print here

File: torchbenchmark/models/mlp/__init__.py [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+        y_train = y_train.reshape(-1, 1)
+        y_val = y_val.reshape(-1, 1)
+
+        train_dataset = TensorDataset(X_train, y_train)
+        test_dataset = TensorDataset(X_val, y_val)
+        self.bs = min(200, X_train.shape[0])
+        self.trainloader = DataLoader(train_dataset, batch_size=self.bs)
+        self.testloader = DataLoader(test_dataset, batch_size=self.bs)
+        self.X_train, self.y_train = next(iter(self.trainloader))
+        self.X_eval, self.y_eval = next(iter(self.testloader))
```
Comment (GeraldCSC) [inline torchbenchmark/models/mlp/__init__.py:47]: is this code necessary for the benchmark? we really just need the first say self.bs rows of X_train and y_val, so perhaps just doing X_train[:self.bs] is sufficient.

File: torchbenchmark/models/mlp/__init__.py [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+        self.X_train, self.y_train = self.X_train.to(self.device), self.y_train.to(self.device)
+        self.X_eval, self.y_eval = self.X_eval.to(self.device), self.y_eval.to(self.device)
```
Comment (GeraldCSC) [inline torchbenchmark/models/mlp/__init__.py:49]: not necessary if you using the suggestion in the comments above

File: torchbenchmark/models/mlp/__init__.py (hunk 4 above)
Comment (adil-a) [reply torchbenchmark/models/mlp/__init__.py:37]: That's pretty cool, didn't know it was a thing in Python :D

File: torchbenchmark/models/mlp/__init__.py [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+
+        self.model = MLP(X_train.shape[1], y_train.shape[1]).to(device)
+        self.optimizer = torch.optim.Adam(self.model.parameters(), lr=lr, weight_decay=weight_decay, eps=eps,
+                                          betas=(beta_1, beta_2))
+        self.criterion = nn.BCELoss()
+
+    def array_to_tensor(self, x):
+        return torch.from_numpy(x.astype(np.float32)).reshape(-1, 1).to(self.device)
+
+    def train(self, niter=1):
```
Comment (GeraldCSC) [inline torchbenchmark/models/mlp/__init__.py:45]: i would guard this against self.jit = True like the get_module function

File: torchbenchmark/models/mlp/__init__.py [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+        self.model.train()
+        for _ in range(niter):
+            out = self.model(self.X_train)
+            self.optimizer.zero_grad()
+            loss = self.criterion(out, self.y_train)
+            loss.backward()
+            self.optimizer.step()
+
+    def eval(self, niter=1):
+        self.model.eval()
```
Comment (GeraldCSC) [inline torchbenchmark/models/mlp/__init__.py:55]: same as above

File: torchbenchmark/models/mlp/__init__.py [hunk 10]
```diff
@@ -0,0 +1,71 @@
...
+
+    def get_module(self):
+        if self.jit:
+            raise NotImplementedError()
+        return self.model, self.X_train
+
+
+if __name__ == '__main__':
+    for device in ['cpu', 'cuda']:
+        for jit in [False, True]:
```
Comment (GeraldCSC) [inline torchbenchmark/models/mlp/__init__.py:67]: no need to test jit=False, since we'd be erroring

**Review by GeraldCSC — APPROVED** [review]
After fixing the jit=True on train and eval please feel free to merge. The rest looks good to me, thanks for this!

**Review by GeraldCSC — APPROVED** [review]

**Pull request merged**